src/System.Windows.Forms.Design/tests/UnitTests/System/Windows/Forms/Design/FlowLayoutPanelDesignerTests.cs

```diff
@@ -0,0 +1,220 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections;
+using System.ComponentModel;
+using System.ComponentModel.Design;
+using System.Drawing;
+using Moq;
+
+namespace System.Windows.Forms.Design.Tests;
+
+public class FlowLayoutPanelDesignerTests : IDisposable
+{
+    private readonly Mock<IServiceProvider> _serviceProviderMock;
+    private readonly Mock<IDesignerHost> _designerHostMock;
+    private readonly Mock<ISelectionService> _selectionServiceMock;
+    private readonly FlowLayoutPanelDesigner _designer;
+    private readonly FlowLayoutPanel _flowLayoutPanel;
+
+    public FlowLayoutPanelDesignerTests()
+    {
+        _serviceProviderMock = new();
+        _designerHostMock = new();
+        _selectionServiceMock = new();
+
+        _serviceProviderMock.Setup(s => s.GetService(typeof(IDesignerHost))).Returns(_designerHostMock.Object);
+        _serviceProviderMock.Setup(s => s.GetService(typeof(ISelectionService))).Returns(_selectionServiceMock.Object);
+
+        _flowLayoutPanel = new() { Site = new Mock<ISite>().Object };
+        _designer = new();
+        _designer.Initialize(_flowLayoutPanel);
+    }
+
+    public void Dispose()
+    {
+        _flowLayoutPanel.Dispose();
+        _designer.Dispose();
```

Comment:
```suggestion
        _designer.Dispose();
        _flowLayoutPanel.Dispose();
```

This way we are guaranteed that control is still available when we are disposing the designer.

Reply:
After changing the dispose order, several tests will be failed. Investigating.

Reply:
Resolved